src/cogent3/core/sequence.py

```diff
@@ -800,7 +801,7 @@ def __init__(
         self.name = name
         orig_seq = seq
         if isinstance(seq, Sequence):
-            seq = seq._seq
+            seq = seq._seq.value
```

Comment:
use `str(self)`

Reply:
in this case, using `str(self)` causes a large amount of tests to fail. It works when using `seq = str(seq._seq)`, is that okay or should it be the former?

Reply:
I think `SequenceI.__str__` is the method being invoked here, which is returning `self._seq`, which is a `SeqView` instance now.

So modify `SequenceI.__str__` to do the following
```python
return str(self._seq)
```

Reply:
with `SequenceI.__str__` modified as you suggest, there still is an issue with  `DnaSequence` objects, which do not have the `._seq` attribute, so when 
``` python
   if isinstance(seq, Sequence):
            seq = str(self)
```
is called, which invokes the `SequenceI.__str__` method, the following error occurs: 
```python 
    def __str__(self):
        """__str__ returns self._seq unmodified."""
>       return str(self._seq)
E       AttributeError: 'DnaSequence' object has no attribute '_seq'
```

Reply:
The following is a quick fix

```python
if isinstance(seq, Sequence):
    try:
        seq = str(self)
    except AttributeError:
        seq = str(seq)
``` 

alternately could make changes to the way in which the `AttributeError` occurs. The call stack from failed tests points to the `ArrayAlignment` to `Sequence` pipeline, via the `to_moltype` method as the problem.